core/src/main/java/org/apache/iceberg/UpdateRequirements.java

```diff
@@ -173,6 +175,27 @@ private void update(MetadataUpdate.SetDefaultSortOrder unused) {
       }
     }
 
+    private void update(MetadataUpdate.RemovePartitionSpecs unused) {
+      // require that the default partition spec has not changed
+      if (!setSpecId) {
+        if (base != null && !isReplace) {
+          require(new UpdateRequirement.AssertDefaultSpecID(base.defaultSpecId()));
+        }
+        this.setSpecId = true;
+      }
+
+      // require that no branches have changed, so that old specs won't be written.
+      if (base != null && !isReplace) {
+        base.refs()
+            .forEach(
+                (name, ref) -> {
+                  if (ref.isBranch() && !name.equals(SnapshotRef.MAIN_BRANCH)) {
+                    require(new UpdateRequirement.AssertRefSnapshotID(name, ref.snapshotId()));
```

Comment:
we should probably also have a test in `TestUpdateRequirements` that actually changes the branch and eventually fails

Reply:
in addition to this, `CatalogTests` currently only tests the happy path, but not the exceptions that are either thrown by `AssertDefaultSpecID` or `AssertRefSnapshotID`. We need to test both cases in `CatalogTests`